.github/workflows/publish-branch-image.yml

```diff
@@ -0,0 +1,51 @@
+name: Publish branch image
```

Comment:
@snwessel and I noticed this is very similar to the `publish` workflow that already exists, can you point out what this one adds that requires a separate workflow, as opposed to being part of the existing workflow?

Reply:
The publish workflow runs automatically on push to dev and at the end triggers a deploy. This workflow runs on demand on branches and does not trigger a deploy. @zarsky-broad offered to clean this up :)